WBooks/Screens/BookLibrary/BookLibraryController.swift

```diff
@@ -61,16 +73,35 @@ extension BookLibraryController: UITableViewDataSource {
     }
     
     func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
-        return libraryItems.count
+        return libraryViewModel.numberOfCells
     }
     
     func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
         
-        let cell: BookCell = tableView.dequeue(cell: BookCell.self)!
-        
-        let book: Book = libraryItems[indexPath.row]
+        guard let cell = tableView.dequeueReusableCell(withIdentifier: "BookCell", for: indexPath) as? BookCell else {
+            fatalError("Cell not exists")
+        }
         
-        cell.configureCell(with: book)
+        cell.bookImage.image = UIImage.bookCover
+        let book = libraryViewModel.getCellViewModel(at: indexPath)
+            let urlString = book.image
+        if let cachedImage = BookInfo.sharedInstance.imageCache.object(forKey: NSString(string: (book.image))) {
```

Comment:
You can add all this logic in the configureCell that you deleted. This way you remove unnecessary responsibilities from the ViewController